test/xml_test.rb

```diff
@@ -95,6 +95,14 @@ def initialize(name=nil)
       it "delegates to #to_node and returns string" do
         assert_xml_equal "<band><name>Rise Against</name></band>", Band.new("Rise Against").to_xml
       end
+
+      describe "encoding option" do
+        it "sets the xml encoding" do
+          representer = Band.new("Die Ärzte")
+          assert_includes representer.to_xml(encoding: "UTF-8"), "Ä"
```

Comment:
I tried to use `assert_xml_equal` here, but it appears that assertion is "too smart" and passes if the two strings represent the same characters, despite having different encodings.

Reply:
Cool, thanks! Can we test that with a nested structure, too?

On a side note: I need help for the rewrite! :smirk:

Reply:
Added a nested structure to the test.  I'm not sure how much help I can be with the rewrite, but I'm happy to take a look :sweat_smile: